### PR TITLE
added support for oauth2 audience

### PR DESCRIPTION
**Pull request by andythsu**

### CODE REVIEW

File: gateway-ha/src/test/java/io/trino/gateway/ha/security/TestLbTokenUtil.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@TestInstance(Lifecycle.PER_CLASS)
+public class TestLbTokenUtil
+{
+    private final String idToken = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ0ZXN0Iiwic3ViIjoidGVzdCIsImF1ZCI6InRlc3QuY29tIn0.YVdTAuEnn-UHsZf5AKA8avaBTLQvdQ3Dd7LeaXKSjlq0atRcmp_IzE8vnJYplYwBs6VNsGpgrfeIMXUsZFKTcfIBinAh05Fj5M7zXVoOPLEdYH5QJ_sfe5-KFPuJs32epC7GE50-umLKTHDZ14dfiXUNK4yAE0bmiNh1lhKcEP35ZAaBt8ldngXJ7BlNbpLpS30-Z4vA6Nht-WjIlluVHd3B5LCJO5FKRcjpuV7E_XZID9c8p0_7Wl7yEvaQB0yJd-VKgT2MQAZse17YqBTb3ix51F5N05s98w7w5zX8NI_afp97YWwNPKTN0dVZFVF8abZ-35O2CAlM0hzbt7NKdw";
```
Comment (willmostly) [inline gateway-ha/src/test/java/io/trino/gateway/ha/security/TestLbTokenUtil.java:34]: Could you use `JWT.create()` to build the token in `setup()`? That would make these tests easier to understand and maintain in the future if new features are desired.

**Review by willmostly — CHANGES_REQUESTED** [review]
Looks good, one improvement requested to the test setup then I think this is good to go

File: gateway-ha/src/test/java/io/trino/gateway/ha/security/TestLbTokenUtil.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+import static org.junit.jupiter.api.Assertions.assertFalse;
```
Comment (mosabua) [inline gateway-ha/src/test/java/io/trino/gateway/ha/security/TestLbTokenUtil.java:30]: Could you refactor this quickly to use assert j ?

**Review by mosabua — APPROVED** [review]
Looks good now.

**Pull request merged**